DataFormats/Math/interface/deltaPhi.h

```diff
@@ -38,7 +44,13 @@ namespace reco {
 
   template <typename T>
   constexpr T deltaPhi(T phi1, T phi2) {
-    return reduceRange(phi1 - phi2);
+    std::cout << std::fixed << std::setprecision(20) << "phi1=" << phi1 << "\nphi2=" << phi2 << '\n';
+    constexpr T epsilon = std::numeric_limits<T>::epsilon();
+    constexpr T diff = phi1 - phi2;
+    if (std::abs(diff) <= epsilon)
+      return (0.);
```

Comment:
I still do not understand the purpose of this.
Here the domain is (-pi,pi), which means that data around zero should be left as is.
This is different from make0To2pi where "epsilon" is supposed to be 2pi-"epsilon" (in quotes because it should be properly normalized to the range, likely 2*numeric_limits<T>::epsilon())

I will need a bit more mental effort to understand the cases at +/-pi because technically only one side should allow exact equality. Is that, perhaps what you were actually aiming to fix?
What is the problematic case?

Reply:
I suspect that deltaPhi is being pass ph1=0. and phi2=0. and I wanted to skip passing the differnece to reduceRange if the difference was just in the last digits of the floats.